TLM/TLM/Custom/AI/CustomResidentAI.cs

```diff
@@ -1,162 +1,203 @@
+    using Manager.Impl;
+    using RedirectionFramework.Attributes;
+    using State;
+    using State.ConfigData;
+    using Traffic.Enums;
+
+    [TargetType(typeof(ResidentAI))]
+    public class CustomResidentAI : ResidentAI {
+        [RedirectMethod]
+        [UsedImplicitly]
+        public string CustomGetLocalizedStatus(ushort instanceId,
+                                               ref CitizenInstance data,
+                                               out InstanceID target) {
+            string ret = Constants.ManagerFactory.ExtCitizenInstanceManager.GetResidentLocalizedStatus(
+                instanceId,
+                ref data,
+                out bool addCustomStatus,
+                out target);
+
+            // NON-STOCK CODE START
+            if (Options.parkingAI && addCustomStatus) {
+                ret = AdvancedParkingManager.Instance.EnrichLocalizedCitizenStatus(
+                    ret,
+                    ref ExtCitizenInstanceManager.Instance.ExtInstances[instanceId],
+                    ref ExtCitizenManager.Instance.ExtCitizens[data.m_citizen]);
+            }
+
+            // NON-STOCK CODE END
+            return ret;
+        }
+
+        [RedirectMethod]
+        [UsedImplicitly]
+        public VehicleInfo CustomGetVehicleInfo(ushort instanceId,
+                                                ref CitizenInstance citizenData,
+                                                bool forceCar,
+                                                out VehicleInfo trailer) {
 #if DEBUG
-			bool citDebug = (GlobalConfig.Instance.Debug.CitizenInstanceId == 0 || GlobalConfig.Instance.Debug.CitizenInstanceId == instanceID) &&
-				(GlobalConfig.Instance.Debug.CitizenId == 0 || GlobalConfig.Instance.Debug.CitizenId == citizenData.m_citizen) &&
-				(GlobalConfig.Instance.Debug.SourceBuildingId == 0 || GlobalConfig.Instance.Debug.SourceBuildingId == citizenData.m_sourceBuilding) &&
-				(GlobalConfig.Instance.Debug.TargetBuildingId == 0 || GlobalConfig.Instance.Debug.TargetBuildingId == citizenData.m_targetBuilding)
-			;
-			bool debug = GlobalConfig.Instance.Debug.Switches[2] && citDebug;
-			bool fineDebug = GlobalConfig.Instance.Debug.Switches[4] && citDebug;
+            bool citizenDebug = (DebugSettings.CitizenInstanceId == 0
+                            || DebugSettings.CitizenInstanceId == instanceId)
+                           && (DebugSettings.CitizenId == 0
+                               || DebugSettings.CitizenId == citizenData.m_citizen)
+                           && (DebugSettings.SourceBuildingId == 0
+                               || DebugSettings.SourceBuildingId == citizenData.m_sourceBuilding)
+                           && (DebugSettings.TargetBuildingId == 0
+                               || DebugSettings.TargetBuildingId == citizenData.m_targetBuilding);
+            bool logParkingAi = DebugSwitch.BasicParkingAILog.Get() && citizenDebug;
+#else
+            var logParkingAi = false;
```

Comment:
Do we need this?

It's used below on line 130. `Log.DebugIf()` won't exist in release builds and the compiler will automatically remove all callee sites which invoke the method.

So there doesn't seem to be any point in having the `#else` and `var logParkingAi = false;` here?

Reply:
The variable has to be defined even if it goes into an deleted function.

Reply:
Does the compiler cull the variable too? If not, it will be having a performance impact on `RELEASE` builds?

Reply:
No there isn't penalty. The "variable" becomes a const, which in Release should lead to culling the code block where `if (false)` is found. We can verify with a decompiler/IL browser.

Reply:
It might be worth checking that as the parking AI gets hammered: Once per vehicle and then possibly several times more if it can't find parking and has to look for another spot.

Reply:
In release there is no code, log call is gone
Highlighted is the IF operator, and second section is the assignment below the Log
The call to Log is gone
Checking a normal variable is very cheap
![image](https://user-images.githubusercontent.com/288724/61585829-e3003900-ab65-11e9-8c76-265fafdcd0f7.png)